SilverSolver/assignment5/unknown_alphabet.py

```diff
@@ -0,0 +1,49 @@
+from collections import defaultdict
+
+def dfs(ordered_vertices, edges, color, v):
```

Comment:
This dfs is seriously scary.

I think it needs at least some comments. What does it return and why?

My current guess is topsort order and a color array if it exists or None if the graph has a cycle, but why return them if the function modifies them anyway?

Also imho searching for cycles and topsorting in the same function causes unnecessary confusion. Maybe comments would help, but why not split this stuff into two functions anyway?

Reply:
Ok, I'll add comments soon